pkg/infra/iac2/iac_templates.go

```diff
@@ -0,0 +1,121 @@
+package iac2
+
+import (
+	"context"
+	_ "embed"
+	"github.com/klothoplatform/klotho/pkg/core"
+	"github.com/klothoplatform/klotho/pkg/query"
+	sitter "github.com/smacker/go-tree-sitter"
+	"github.com/smacker/go-tree-sitter/typescript/typescript"
+	"io"
+	"regexp"
+	"strings"
+	"text/template"
+)
+
+type (
+	ResourceCreationSignature struct {
+		inputTypes map[string]string
+		outputType string
+	}
+
+	ResourceCreationTemplate struct {
+		ResourceCreationSignature
+		imports            map[string]struct{}
+		expressionTemplate string
+	}
+)
+
+var (
+	tsLanguage = core.SourceLanguage{
+		ID:     core.LanguageId("ts"),
+		Sitter: typescript.GetLanguage(),
+	}
+
+	parameterizeArgsRegex = regexp.MustCompile(`args(\.\w+)`)
+
+	//go:embed find_args.scm
+	findArgsQuery string
+
+	//go:embed find_create_func.scm
+	findCreateFuncQuery string
+
+	//go:embed find_imports.scm
+	findImportsQuery string
+)
+
+func ParseResourceCreationTemplate(contents []byte) ResourceCreationTemplate {
+	node := parseFile(contents)
+
+	result := ResourceCreationTemplate{}
+
+	// inputs
+	result.inputTypes = make(map[string]string)
+	nextInput := doQuery(node, findArgsQuery)
+	for {
+		match, found := nextInput()
+		if !found {
+			break
+		}
+		inputName, inputType := match["property_name"].Content(), match["property_type"].Content()
+		result.inputTypes[inputName] = inputType
+	}
+
+	// return type
+	returnTypeFunc := query.Select(doQuery(node, findCreateFuncQuery), query.ContentOf(query.ParamNamed("return_type")))
+	returnType, err := returnTypeFunc()
+	if !err {
+		// unexpected, since all inputs are from resources in the klotho binary
+		panic("couldn't find valid create() function")
+	}
+	result.outputType = returnType
+
+	// return expression
+	returnBodyFunc := query.Select(doQuery(node, findCreateFuncQuery), query.ContentOf(query.ParamNamed("return_body")))
+	returnBody, err := returnBodyFunc()
+	if !err {
+		// unexpected, since all inputs are from resources in the klotho binary
+		panic("couldn't find valid create() function")
+	}
+	result.expressionTemplate = parameterizeArgs(returnBody)
+
+	// imports
+	result.imports = make(map[string]struct{})
+	importsQuery := query.Select(doQuery(node, findImportsQuery), query.ContentOf(query.ParamNamed("import")))
+	for _, importLine := range query.Collect(importsQuery) {
+		// Trim any trailing semicolons. This helps normalize imports, so that we don't include them twice if one file
+		// includes the semicolon and the other doesn't.
+		importLine = strings.TrimRight(importLine, ";")
+		result.imports[importLine] = struct{}{}
+	}
+
+	return result
+}
+
+// doQuery is a thin wrapper around `query.Exec` to use typescript as the Language.
+func doQuery(c *sitter.Node, q string) query.NextMatchFunc {
+	return query.Exec(tsLanguage, c, q)
+}
+
+// parameterizeArgs turns "args.foo" into {{.Foo}}. It's very simplistic and just works off regex
+func parameterizeArgs(contents string) string {
```

Comment:
As discussed in our chat, would be nice to use treesitter instead of regexp for this.
Here's what I threw together in the playground during the call in case it's useful:
```
(member_expression
	object: (identifier) @obj
	(#eq @obj "args")
	property: (property_identifier) @prop
 )
```

Reply:
I'll do that early next week (just so we can focus on gluing Jordan's and my work tonight/tomorrow). I've got a todo list locally, so I won't forget!